flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/PendingCheckpoint.java

```diff
@@ -105,12 +105,12 @@
     /** The checkpoint properties. */
     private final CheckpointProperties props;
 
-    /** Target storage location to persist the checkpoint metadata to. */
-    private final CheckpointStorageLocation targetLocation;
-
     /** The promise to fulfill once the checkpoint has been completed. */
     private final CompletableFuture<CompletedCheckpoint> onCompletionPromise;
 
+    /** Target storage location to persist the checkpoint metadata to. */
+    private CheckpointStorageLocation targetLocation;
```

Comment:
Since it is not final anymore we need to carefully think about the thread model here. I mean as I understand this field can be used from several threads. So perhaps it should be volatile or we should use it only under the lock.(I am not sure about that but it should be checked)

Reply:
Thanks for your reminder. 
I will add volatile for targetLocation. And I think disposed also should be volatile, because there isn't use lock in `isDisposed` method. How do you think?